internal/monitors/collectd/solr/solr.go

```diff
@@ -0,0 +1,78 @@
+package solr
+
+import (
+	"github.com/signalfx/signalfx-agent/internal/core/config"
+	"github.com/signalfx/signalfx-agent/internal/monitors"
+	"github.com/signalfx/signalfx-agent/internal/monitors/collectd"
+	"github.com/signalfx/signalfx-agent/internal/monitors/collectd/python"
+	"github.com/signalfx/signalfx-agent/internal/monitors/pyrunner"
+)
+
+const monitorType = "collectd/solr"
+
+// MONITOR(collectd/solr): Monitors Solr instances.
+//
+// See https://github.com/signalfx/collectd-solr and
+// https://github.com/signalfx/integrations/tree/master/collectd-solr
+
+func init() {
+	monitors.Register(monitorType, func() interface{} {
+		return &Monitor{
+			python.PyMonitor{
+				MonitorCore: pyrunner.New("sfxcollectd"),
+			},
+		}
+	}, &Config{})
+}
+
+// Config is the monitor-specific config with the generic config embedded
+type Config struct {
+	config.MonitorConfig `yaml:",inline" acceptsEndpoints:"true"`
+	pyConf               *python.Config
+	Host                 string `yaml:"host" validate:"required"`
+	Port                 uint16 `yaml:"port" validate:"required"`
+	// Cluster name of this solr cluster.
+	Cluster string `yaml:"cluster"`
+	// EnhancedMetrics boolean to indicate whether stats from /metrics are needed
+	EnhancedMetrics *bool `yaml:"enhancedMetrics" default:"false"`
+	// IncludeMetric metric name from the /admin/metrics endpoint to include (valid when EnhancedMetrics is "false")
+	IncludeMetric string `yaml:"includeMetric"`
```

Comment:
This will only allow for one metric to include since yaml doesn't support repeated keys like the collectd config format does.  You need to make this a `[]string` and then make the value passed to collectd via the `PluginConfig` map be this:

```
"IncludeMetric": {
  "#flatten": true,
  "values": conf.IncludeMetric,
}
```

Reply:
If I just change conf.IncludeMetric to []string as you suggest, why won't "IncludeMetric": conf.IncludeMetric work?

Reply:
Because the collectd plugin doesn't support arrays/lists for the IncludeMetric config option.  It expects a bunch of separate instances of IncludeMetric on separate lines.

Reply:
Thanks nvm. I'm looking at what you did for redis and I understand.